src/test/scala/com/fulcrumgenomics/bam/SplitBamByReadGroupTest.scala

```diff
@@ -0,0 +1,97 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2017 Fulcrum Genomics LLC
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+package com.fulcrumgenomics.bam
+
+import java.nio.file.Files
+
+import com.fulcrumgenomics.commons.CommonsDef._
+import com.fulcrumgenomics.testing.{SamRecordSetBuilder, UnitSpec}
+import htsjdk.samtools.SAMRecordSetBuilder
+
+class SplitBamByReadGroupTest extends UnitSpec {
+
+  private def newOutput = Files.createTempDirectory("split_bam_by_read_group_test").resolve("prefix")
+
+  // HACK!
+  private val defaultRgId: String = new SAMRecordSetBuilder().getHeader.getReadGroups.map(_.getId).toSeq.head
+  private val otherRgId: String = "2"
+
+  // Make a set of records to run multiple tests on
+  private val builder = new SamRecordSetBuilder(readGroupId=Some(otherRgId))
+  builder.addPair("rg_default_1", 0, 100, 300).foreach(_.setAttribute("RG", defaultRgId))
```

Comment:
Also, it'll make moving this to the new builder much easier if you name the parameters please.  I.e. 
```scala
builder.addPair("rg_default_1", 0, 100, 300).foreach(_.setAttribute("RG", defaultRgId))
->
builder.addPair(name="rg_default_1", contig=0, start1=100, start2=300, attrs=Map("RG" -> defaultRgId))
```